From a5e8efbe498066344b7ea45eb1a82484cbdc5dde Mon Sep 17 00:00:00 2001
From: Amelie Delaunay <amelie.delaunay@foss.st.com>
Date: Thu, 1 Feb 2024 15:19:37 +0100
Subject: [PATCH 0995/1141] dt-bindings: dma: stm32-dma3: introduce
 memory-region property

DMA3 requires memory to allocate its linked-list items (LLIs). By default,
when memory-region property is not specified, LLIs are allocated in DDR.
Thanks to memory-region property, a dedicated /reserved-memory child node
can be used for LLIs allocation. It is then recommended to use internal
memory: LLI read for registers update is optimal from internal RAM.

Change-Id: Id809c315d8d3027d8f49a8a7b9e4a14c6268eecb
Signed-off-by: Amelie Delaunay <amelie.delaunay@foss.st.com>
Reviewed-on: https://gerrit.st.com/c/mpu/oe/st/linux-stm32/+/358988
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
Reviewed-by: Antonio Maria BORNEO <antonio.borneo@foss.st.com>
Domain-Review: Fabrice GASNIER <fabrice.gasnier@foss.st.com>
---
 .../devicetree/bindings/dma/stm32/st,stm32-dma3.yaml     | 9 +++++++++
 1 file changed, 9 insertions(+)

diff --git a/Documentation/devicetree/bindings/dma/stm32/st,stm32-dma3.yaml b/Documentation/devicetree/bindings/dma/stm32/st,stm32-dma3.yaml
index 1ba09cfbd30c..0309848fa308 100644
--- a/Documentation/devicetree/bindings/dma/stm32/st,stm32-dma3.yaml
+++ b/Documentation/devicetree/bindings/dma/stm32/st,stm32-dma3.yaml
@@ -91,6 +91,15 @@ properties:
     minItems: 4
     maxItems: 16
 
+  memory-region:
+    description: |
+      Phandle to a /reserved-memory child node assigned to STM32 DMA3 for linked-list items
+      allocation. When not defined, linked-list items are allocated in DDR.
+      During transfer, STM32 DMA3 updates its registers reading the linked-list items in memory.
+      There reads are 32-bit single read. Allocating the linked-list items in internal RAM improves
+      the STM32 DMA3 performance.
+    maxItems: 1
+
   power-domains:
     maxItems: 1
 
-- 
2.39.2

